[donet] Update dotnet documentation for 2.22 release (#16464)
DataDog/documentation: content/en/profiler/enabling/dotnet.md

.NET 6<br/>
.NET 7
<div class="alert alert-warning">

>   <strong>Note:</strong> The beta for lock contention is only available for .NET 5+, only in .NET 6+ for allocation profiling and .NET 7+ is required for live heap profiling.

Reviewer: ```suggestion
  <strong>Note:</strong><br />Lock Contention profiling beta is only available for .NET 5+. <br />Allocations profiling beta is only for .NET 6+.<br />Live Heap profiling beta is only for .NET 7+.
```
